File: t-tasks/src/main/java/com/teo/ttasks/ui/items/TaskListItem.java
package com.teo.ttasks.ui.items;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.view.View;

import com.mikepenz.fastadapter.FastAdapter;
import com.mikepenz.fastadapter.items.AbstractItem;
import com.teo.ttasks.R;
import com.teo.ttasks.data.model.TTaskList;
import com.teo.ttasks.databinding.ItemTaskListBinding;

import java.util.List;

import timber.log.Timber;

public class TaskListItem extends AbstractItem<TaskListItem, TaskListItem.ViewHolder> {

    private final TTaskList taskList;
    private final long taskCount;

    public TaskListItem(TTaskList taskList, long taskCount) {
        this.taskList = taskList;
        this.taskCount = taskCount;
        Timber.d("count %d", taskCount);
    }

    public String getTitle() {
        return taskList.getTitle();
    }

    public String getId() {
        return taskList.getId();
    }

    long getTaskCount() {
        return taskCount;
    }

    @Override
    public int getType() {
        return R.id.task_list_item;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.item_task_list;
    }

    @Override public ViewHolder getViewHolder(@NonNull View view) {
        return new ViewHolder(view);
    }

    public static class ViewHolder extends FastAdapter.ViewHolder<TaskListItem> {

        public final ItemTaskListBinding itemTaskListBinding;

        ViewHolder(View view) {
            super(view);
            itemTaskListBinding = DataBindingUtil.bind(view);
        }

        @Override
        public void bindView(@NonNull TaskListItem item, @NonNull List<Object> payloads) {
            Context context = itemTaskListBinding.getRoot().getContext();
            long taskCount = item.getTaskCount();
            itemTaskListBinding.taskListSize.setText(taskCount > 0 ?
                    context.getString(R.string.task_list_size, taskCount) :
                    context.getString(R.string.empty_task_list));
            itemTaskListBinding.taskListTitle.setText(item.getTitle());
        }

        @Override
        public void unbindView(@NonNull TaskListItem item) {
            itemTaskListBinding.taskListSize.setText(null);
            itemTaskListBinding.taskListTitle.setText(null);
        }
    }
}

File: t-tasks/src/main/java/com/teo/ttasks/ui/items/TaskListItem.kt
package com.teo.ttasks.ui.items

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.teo.ttasks.R
import com.teo.ttasks.data.model.TaskList
import com.teo.ttasks.databinding.ItemTaskListBinding
import eu.davidea.flexibleadapter.FlexibleAdapter
import eu.davidea.flexibleadapter.items.AbstractFlexibleItem
import eu.davidea.flexibleadapter.items.IFlexible
import eu.davidea.viewholders.FlexibleViewHolder
import timber.log.Timber

class TaskListItem(private val taskList: TaskList,
                   private val taskCount: Long) : AbstractFlexibleItem<TaskListItem.ViewHolder>() {

    init {
        Timber.d("count %d", taskCount)
    }

    val title: String
        get() = taskList.title

    val id: String
        get() = taskList.id

    override fun getLayoutRes(): Int {
        return R.layout.item_task_list
    }

    override fun bindViewHolder(adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>,
                                viewHolder: ViewHolder, position: Int, payloads: MutableList<Any?>) {
        val itemTaskBinding = viewHolder.itemTaskListBinding
        val context = itemTaskBinding.root.context
        val taskCountInt = taskCount.toInt()

        itemTaskBinding.taskListTitle.text = taskList.title
        itemTaskBinding.taskListSize.text =
                if (taskCountInt > 0) {
                    context.resources.getQuantityString(R.plurals.task_list_size, taskCountInt, taskCountInt)
                } else {
                    context.getString(R.string.empty_task_list)
                }
        itemTaskBinding.deleteTaskList.setOnClickListener(viewHolder)
    }

    override fun createViewHolder(view: View, adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>): ViewHolder {
        return ViewHolder(view, adapter)
    }

    override fun equals(other: Any?): Boolean = other is TaskListItem && id == other.id

    override fun hashCode(): Int {
        var result = taskList.hashCode()
        result = 31 * result + taskCount.hashCode()
        return result
    }

    class ViewHolder internal constructor(view: View, adapter: FlexibleAdapter<out IFlexible<*>>) : FlexibleViewHolder(view, adapter) {
        var itemTaskListBinding = ItemTaskListBinding.bind(view)
    }
}
